### PR TITLE
[login][request_account] Adding project field to request account

### PR DESCRIPTION
**Pull request by jesscall**
## Brief summary of changes
This PR adds `project` field to the request account form.

- [ ] Have you updated related documentation? **NOT NEEDED**.

#### Testing instructions (if applicable)

1. Follow the test plan for request account and assure that the bugs reported during testing round have been amended.

#### Link(s) to related issue(s)
* Resolves #  (Reference the issue this fixes, if any.)
#6233 
#6245

### CODE REVIEW

File: modules/login/php/requestaccount.class.inc [hunk 1]
```diff
@@ -171,6 +181,14 @@ class RequestAccount extends \NDB_Form
                 )
             );
 
+            $DB->insert(
+                'user_project_rel',
+                array(
+                    'UserID'    => $user_id,
+                    'ProjectID' => $site,
```
Comment (ridz1208) [inline modules/login/php/requestaccount.class.inc:188]:
```suggestion
                    'ProjectID' => $project,
```

**Comment by ridz1208** [comment]
@maltheism please test again... but better

**Comment by maltheism** [comment]
@ridz1208 🥺 ..but the manual test resolved all the issues. 😭 I failed as a tester.

**Comment by maltheism** [comment]
@ridz1208 I just manually tested this PR and the issues the PR resolves are solved. I'm still confused why the issues were resolved with the previous code mistake.

**Comment by ridz1208** [comment]
@maltheism it was still putting a value in the table but instead of putting the project ID value
it was putting the site ID in the project column lol

File: modules/login/php/requestaccount.class.inc [hunk 2]
```diff
@@ -171,6 +181,14 @@ class RequestAccount extends \NDB_Form
                 )
             );
 
+            $DB->insert(
+                'user_project_rel',
+                array(
+                    'UserID'    => $user_id,
+                    'ProjectID' => $project,
```
Comment (driusan) [inline modules/login/php/requestaccount.class.inc:188]: `$project` came from the value submitted by the user. It needs to be validated by the backend before being inserted and shouldn't be blindly trusted.

Comment (ridz1208) [reply modules/login/php/requestaccount.class.inc:188]: then same should be done with site

Comment (driusan) [reply modules/login/php/requestaccount.class.inc:188]: True, might as well do that at the same time.

Comment (jesscall) [reply modules/login/php/requestaccount.class.inc:188]: @driusan what kind of validation would you like to see here? The values for site and project are coming from drop-downs populated with values from the DB.

Comment (ridz1208) [reply modules/login/php/requestaccount.class.inc:188]: they are then passed to the insert query as parameters and inserted into a table with foreign keys...

Comment (driusan) [reply modules/login/php/requestaccount.class.inc:188]: 1. The value submitted is an integer.
2. The value integer submitted exists as a key in the appropriate (project/psc) tables.

**Pull request merged**